---
title: "mypopgen Package Vignette"
author: "Matthew Murray"
date: "2015-05-07"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{mypopgen Package}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8](inputenc)
---

A vignette for the use of the mypopgen package. Illistrated with made up data. 
```{r}
library("mypopgen")
```

Creating some data to play with. 

```{r}
bases<- c("A","T","C","G")
seq1 <- sample(bases,1000, replace=TRUE)
seqs<- as.data.frame(matrix(NA,nrow=9,ncol=1000))
seqs[1,]<-seq1
for ( i in 2:10){
snps<- sample(bases,3, replace=TRUE)
snp_pos<-sample(seq(1,1000),3)
seq2<- seq1
seq2[snp_pos]<-snps
seqs[i,]<-seq2
}
```
 Take a look at the theoretical 1 kb of sequence created
```{r Sequence}
seq1
```

Here is a snippit of multiple samples, somehting similar to a potential real dataset
```{r}
seqs[1:10,1:20]
```

Now we can find which sites in our kb or sequence using the "my_seg_sites" function
```{r my_seg_sites}
my_seg_sites(seqs)
```

Now if we pull out the positions that are segregating to take a look. We can see all of these sites have at least one snp. 
```{r, results='hold'}
sites<- my_seg_sites(seqs)
seqs[,sites]
```

Now we can take all of these sites and calculate nucleotide diversity using the "my_pi" funciton. 
```{r}
my_pi(seqs)
```

Now look at Fst between pairs of markers. 

First we create a second "population" to test. Here we create a two loci that are segregating, and create two population that start at the same frequency and then move away. 
```{r}
m1_p1<- data.frame(sample(c(rep("A",50),rep("C",50))))
m1_p2<- data.frame(sample(c(rep("A",50),rep("C",50))))
m2_p1<- data.frame(sample(c(rep("A",50),rep("C",50))))
m2_p2<- data.frame(sample(c(rep("A",50),rep("C",50))))
```

Now we need to calculate major allele frequency, we can do this by using the "my_allele_freq" function. This gives us what we would expect, an allele frequency of 0.5 for wach site. 
```{r allele freq}
my_allele_freq(m1_p1)
freq_1_1= my_allele_freq(m1_p1)
freq_1_2= my_allele_freq(m1_p2)
freq_2_1= my_allele_freq(m2_p1)
freq_2_2= my_allele_freq(m2_p2)
```


Calculate the fst between populations at each snp 
```{r Fst}
my_Fst(2, c(50,50), c(freq_1_1,freq_1_2))
my_Fst(2, c(50,50), c(freq_2_1,freq_2_2))
```
We can see here that the two population have the same allele frequencies and are not two seperate populations, based on these markers at this time. 

Now if we apply some type of selcetion at one of the loci, we can see the Fst shift. 
```{r}
m1_p1<- data.frame(sample(c(rep("A",50),rep("C",50))))
m1_p2<- data.frame(sample(c(rep("A",50),rep("C",50))))
m2_p1<- data.frame(sample(c(rep("A",50),rep("C",50))))
m2_p2<- data.frame(sample(c(rep("A",50),rep("C",50))))

selects<-sample(which(m1_p1 == "A"), size= length(which(m1_p1 == "A"))/3)
m1_p1[selects,]<-"C"


freq_1_1= my_allele_freq(m1_p1)
freq_1_2= my_allele_freq(m1_p2)
freq_2_1= my_allele_freq(m2_p1)
freq_2_2= my_allele_freq(m2_p2)

#Marker 1
my_Fst(2, c(50,50), c(freq_1_1,freq_1_2))
#Marker 2
my_Fst(2, c(50,50), c(freq_2_1,freq_2_2))
```
After this simulated selection, the Fst at marker1 has changed, and gone up, so the two popultaions are now more different than they were before. 

